File: application/src/main/java/com/example/server/repository/entity/ItemRequest.java
package com.example.server.repository.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@AllArgsConstructor
@Entity
@EqualsAndHashCode(of = "id")
@Getter
@NoArgsConstructor
@Setter
@SequenceGenerator(name = "requests_id_seq", allocationSize = 1)
@Table(name = "requests")
public class ItemRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "requests_id_seq")
    private Long id;

    @Column(name = "description")
    private String description;

    @JoinColumn(name = "requester_id")
    @ManyToOne
    private User requester;

    @Column(name = "created")
    @CreationTimestamp
    private LocalDateTime created;
}

File: application/src/main/kotlin/com/example/server/repository/entity/ItemRequest.kt
package com.example.server.repository.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Entity
@SequenceGenerator(name = "requests_id_seq", allocationSize = 1)
@Table(name = "requests")
class ItemRequest (

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "requests_id_seq")
    var id: Long?,

    @Column(name = "description")
    var description: String,

    @JoinColumn(name = "requester_id")
    @ManyToOne
    var requester: User,

    @Column(name = "created")
    @CreationTimestamp
    var created: LocalDateTime?
)
